docs/flow/reference/components/discovery.kubernetes.md

```diff
@@ -0,0 +1,163 @@
+[`tls_config`](#tls_config-block) | Configure TLS settings for connecting to targets | no
+
+#### `basic_auth` block
+
+Name          | Type     | Description                                     | Default | Required
+------------- | -------- | ----------------------------------------------- | ------- | -------
+`username`      | `string`   | Setup of Basic HTTP authentication credentials. |         | no
+`password`      | `secret`   | Setup of Basic HTTP authentication credentials. |         | no
+`password_file` | `string`   | Setup of Basic HTTP authentication credentials. |         | no
+
+#### `authorization` block
+
+Name                  | Type       | Description                              | Default | Required
+--------------------- | ---------- | ---------------------------------------- | ------- | --------
+`type`                | `string`   | Setup of HTTP Authorization credentials. |         | no
+`credential`          | `secret`   | Setup of HTTP Authorization credentials. |         | no
+`credentials_file`    | `string`   | Setup of HTTP Authorization credentials. |         | no
+
+#### `oauth2` block
+
+Name                 | Type                 | Description                              | Default | Required
+-------------------- | -------------------- | ---------------------------------------- | ------- | --------
+`client_id`          | `string`             | Setup of the OAuth2 client.              |         | no
+`client_secret`      | `secret`             | Setup of the OAuth2 client.              |         | no
+`client_secret_file` | `string`             | Setup of the OAuth2 client.              |         | no
+`scopes`             | `list(string)`       | Setup of the OAuth2 client.              |         | no
+`token_url`          | `string`             | Setup of the OAuth2 client.              |         | no
+`endpoint_params`    | `map(string)`        | Setup of the OAuth2 client.              |         | no
+`proxy_url`          | `string`             | Setup of the OAuth2 client.              |         | no
+
+The `oauth2` block may also contain its own separate `tls_config` sub-block.
+
+#### `tls_config` block
+
+Name                              | Type       | Description                                | Default | Required
+--------------------------------- | ---------- | ------------------------------------------ | ------- | --------
+`tls_config_ca_file`              | `string`   | Configuration options for TLS connections. |         | no
+`tls_config_cert_file`            | `string`   | Configuration options for TLS connections. |         | no
+`tls_config_key_file`             | `string`   | Configuration options for TLS connections. |         | no
+`tls_config_server_name`          | `string`   | Configuration options for TLS connections. |         | no
+`tls_config_insecure_skip_verify` | `bool`     | Configuration options for TLS connections. |         | no
```

Comment:
Can you copy the documentation for basic_auth, authorization, oauth2, and tls_config from [prometheus.remote_write](https://github.com/grafana/agent/blob/main/docs/flow/reference/components/prometheus.remote_write.md#basic_auth-block) instead? It generally provides more information.